### ISSUE TITLE
Update Dockerfile to move to python 3.8

### ISSUE THREAD

**Issue by nurmi** [anchore/anchore-engine#605]
In current (0.8.X) series, engine is using python36 environment from UBI8.  Noting here that the python36 series is set to EOL toward the end of 2021

https://www.python.org/dev/peps/pep-0494/#lifespan

Updates to the Dockerfile are minimal (python, pip, wheel) should be updated to move to python38

**Comment by keenan-v1** [comment]
Note that I've just recently migrated to Python 3.7 and had issues with policy-engine referencing `typing.GenericMeta` which has been removed. The issue does not show up in build, but when running and accessing things via the API.

Example:
`$ anchore-cli image add registry.docker/some:image` works. The scan completes.
`$ anchore-cli image vuln registry.docker/some:image all` fails with a 500.

The logs contain the stack trace but often buried. The specific file is https://github.com/anchore/anchore-engine/blob/master/anchore_engine/services/policy_engine/api/util.py#L26.

For now, I did some debugging and found some helpful code on Stack Overflow here: https://stackoverflow.com/questions/61918932/attributeerror-module-typing-has-no-attribute-genericmeta which allowed me to fix the issue with minimal changes to the code base.
